backend/controller/lmevaljob_controller.go

```diff
@@ -186,40 +237,118 @@ func (r *LMEvalJobReconciler) SetupWithManager(mgr ctrl.Manager) error {
+		return fmt.Errorf("JobName or JobNameSpace is empty")
 	}
-	if v, found := configmap.Data[ImagePullPolicyKey]; found {
-		if p, found := pullPolicyMap[corev1.PullPolicy(v)]; found {
-			r.options.ImagePullPolicy = p
-		} else {
-			log.Error(
-				fmt.Errorf("invalid %s value in the configmap: %s", ImagePullPolicyKey, v),
-				"use the default value for the ImagePullPolicy instead",
-			)
+
+	job := &lmevalservicev1beta1.LMEvalJob{}
+	if err := r.Get(ctx, types.NamespacedName{
+		Namespace: newStatus.JobNamespace,
+		Name:      newStatus.JobName,
+	}, job); err != nil {
+		log.Info("unable to fetch LMEvalJob")
+		return err
+	}
+
+	job.Status.State = lmevalservicev1beta1.JobState(newStatus.GetState())
+	job.Status.Reason = lmevalservicev1beta1.Reason(newStatus.GetReason())
+	if newStatus.GetStatusMessage() != "" {
+		job.Status.Message = newStatus.GetStatusMessage()
+	}
+	if newStatus.Results != nil {
+		job.Status.Results = newStatus.GetResults()
+	}
+
+	err := r.Status().Update(ctx, job)
+	if err != nil {
+		log.Error(err, "failed to update status")
+	}
+	return err
+}
+
+func (r *LMEvalJobReconciler) constructOptionsFromConfigMap(
+	ctx context.Context, configmap *corev1.ConfigMap) error {
+	r.options = &ServiceOptions{
+		DriverImage:          DefaultDriverImage,
+		PodImage:             DefaultPodImage,
+		DriverServiceAccount: DefaultDriverServiceAccount,
+		PodCheckingInterval:  DefaultPodCheckingInterval,
+		ImagePullPolicy:      DefaultImagePullPolicy,
+		GrpcPort:             DefaultGrpcPort,
+		GrpcService:          DefaultGrpcService,
+		GrpcServerSecret:     DefaultGrpcServerSecret,
+		GrpcClientSecret:     DefaultGrpcClientSecret,
+	}
+
+	log := log.FromContext(ctx)
+	rv := reflect.ValueOf(r.options).Elem()
+	var msgs []string
+
+	for idx, cap := 0, rv.NumField(); idx < cap; idx++ {
+		frv := rv.Field(idx)
+		fname := rv.Type().Field(idx).Name
+		configKey, ok := optionKeys[fname]
+		if !ok {
+			continue
+		}
+
+		if v, found := configmap.Data[configKey]; found {
+			var err error
+			switch frv.Type().Name() {
+			case "string":
+				frv.SetString(v)
+			case "int":
+				var grpcPort int
+				grpcPort, err = strconv.Atoi(v)
+				if err == nil {
+					frv.SetInt(int64(grpcPort))
+				}
+			case "Duration":
+				var d time.Duration
+				d, err = time.ParseDuration(v)
+				if err == nil {
+					frv.Set(reflect.ValueOf(d))
+				}
+			case "PullPolicy":
+				if p, found := pullPolicyMap[corev1.PullPolicy(v)]; found {
+					frv.Set(reflect.ValueOf(p))
+				} else {
+					err = fmt.Errorf("invalid PullPolicy")
+				}
+			default:
+				err = fmt.Errorf("can not handle the config %v, type: %v", optionKeys[fname], frv.Type().Name())
+			}
+
+			if err != nil {
+				msgs = append(msgs, fmt.Sprintf("invalid setting for %v: %v, use default setting instead", optionKeys[fname], v))
+			}
 		}
 	}
+
+	if len(msgs) > 0 {
+		log.Error(fmt.Errorf("some settings in the configmap are invalid"), strings.Join(msgs, "\n"))
```

Comment:
since we use default values for those invalid settings, it only logs error messages here. but I am thinking make the `default` case to error out. in that case, we can catch unsupported data types in the `ServiceOptions` struct. (although it shouldn't happen haha)

Reply:
I throw an error for the `default` case.